api/proto/config/storage.proto

```diff
@@ -242,3 +242,39 @@ message DisksConfig {
   DisksArrayType array_type = 2;
   repeated DisksConfig children = 3;
 }
+
+// RollbackCmd - snapshot rollback command
+message RollbackCmd {
+  string snpshot_uuid = 1;
+  string volume_uuid = 2;
+  DeviceOpsCmd rollback = 3;
+}
+
+// SnapshotConfig describes a snapshot for a specific logical
+// volume that must exist on the device. It has a required field
+// volume_uuid for which it was created and UUID for snapshot name in ZFS.
+message SnapshotConfig {
+  // The real name of the snapshot in ZFS. It is the link between
+  // the command and the response to the command. It is assumed
+  // that the field will always be filled with a unique value
+  // that the controller will generate.
+  string uuid = 1;
+  // Display name (User-friendly name). This name does not affect
+  // the snapshot properties in ZFS in any way.
+  // Can be filled on the controller side.
+  // If a snapshot has already been created and this field has changed,
+  // it can be assumed that the friendly name for this snapshot has
+  // been renamed on the controller side.
+  string display_name = 2;
+  // Volume ID of the volume this snapshot belongs to. The field is
+  // required for all messages. Must always be filled in on the
+  // controller side before sending to create snapshot command.
+  string volume_uuid = 3;
+  // The command for the snapshot rollback operation.
```

Comment:
Add "optional" to make it "An optional command ..." this not all SnapshotConfig has a rollback.

I wonder whether we should make this more explicit in the API by adding a boolean has_rollback, or add a comment specifying that if no RollbackCmd message is included it means there is no rollback requested. The latter makes more sense; it is easy to check whether the embedded message is included or nil when processing the SnapshotCmd.

Reply:
What do you mean by having a fallback?
Why shouldn't it be required? I believe we should always compare counters to determine a rollback operation.🤔

Reply:
I didn't say "fallback". I said "rollback".

Basically the way the message is set up it can be misunderstood as each snapshotconfig always has a rollbackcmd, when the rollbackcmd is in fact optional and only used when there is a rollback operation.

Reply:
Corrected and updated a comment